Bottom: 6f3a6a83f563c853be17327b0aed9323f19cc868
Top:    0db9bbc0b5a0dd19272db4a464ceef5ca6a0b191
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-03-23 00:02:23 +0000

Tidy up range check in misc.c

For consistnecy with other range checks involving max_copper_layer, and to
drive home the fact we are checking for exceeding max_copper_layer + 2.
NB: The +2 relates to the bottom silk and top-silk layers, whose numbers
are used to denote the bottom and top layer groups respectively.

re-write:
  layer > max_copper_layer + 1

to become:
  layer >= max_copper_layer + 2


---

diff --git a/src/hid/gtk/gui-config.c b/src/hid/gtk/gui-config.c
index eb8fbd1..0c48026 100644
--- a/src/hid/gtk/gui-config.c
+++ b/src/hid/gtk/gui-config.c
@@ -1513,6 +1513,15 @@ config_layer_group_button_state_update (void)
   for (g = 0; g < max_group; g++)
     for (i = 0; i < layer_groups.Number[g]; i++)
       {
+	/* Skip anything beyond the bottom silk / top silk layers residing
+	   at index max_copper_layer + 0 and max_copper_layer + 1. These are
+	   used to indicate the groups for bottom and top sides respectively,
+	   so form somewhat of a special case in our handling here (they have
+	   group buttons).
+	 */
+	if (layer_groups.Entries[g][i] >= max_copper_layer + 2)
+	  continue;
+
 /*			printf("layer %d in group %d\n", layer_groups.Entries[g][i], g +1); */
 	config_layer_group[layer_groups.Entries[g][i]] = g + 1;
 	gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON
diff --git a/src/misc.c b/src/misc.c
index 4f381ae..37aaebb 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -2015,7 +2015,7 @@ MoveLayerToGroup (int layer, int group)
 {
   int prev, i, j;
 
-  if (layer < 0 || layer > max_copper_layer + 1)
+  if (layer < 0 || layer >= max_copper_layer + 2)
     return -1;
   prev = GetLayerGroupNumberByNumber (layer);
   if ((layer == bottom_silk_layer
